Clarify previous parameter expression (#3214)
modelica/ModelicaSpecification: chapters/synchronous.tex

\section{Clocked State Variables}\label{clocked-state-variables}

\begin{definition}[Clocked state variable]\label{def:clocked-state-variable}\index{clocked!state variable}\index{state variable!clocked}
A component expression which is not a parameter, and to which \lstinline!previous! has been applied.
\end{definition}

The previous value of a clocked variable can be accessed with the \lstinline!previous! operator, listed below.
\begin{center}
\begin{tabular}{l|l l}
\hline
\tablehead{Expression} & \tablehead{Description} & \tablehead{Details}\\
\hline
\hline
{\lstinline!previous($u$)!} & Previous value of clocked variable & \Cref{modelica:previous} \\
\hline
\end{tabular}
\end{center}

\begin{operatordefinition}[previous]
\begin{synopsis}\begin{lstlisting}
previous($u$)
\end{lstlisting}\end{synopsis}
\begin{semantics}
The input argument $u$ is a component expression (\cref{def:component-expression}).

> If $u$ is a parameter its value is returned.

Reviewer: Usually, a comma is recommended here:
```suggestion
If $u$ is a parameter, its value is returned.
```

\end{tabular}
\end{center}

> A variable (i.e. not parameter) to which \lstinline!previous! has been applied is called a \firstuse[clocked!state variable]{clocked state variable}\index{state variable!clocked}.

Reviewer 2: Unguarded in-sentence period:
```suggestion
A variable (i.e., not parameter) to which \lstinline!previous! has been applied is called a \firstuse[clocked!state variable]{clocked state variable}\index{state variable!clocked}.
```

Reviewer 2: This comment is now obsolete.  Resolving.

\begin{operatordefinition}[previous]
\begin{synopsis}\begin{lstlisting}
previous($u$)
\end{lstlisting}\end{synopsis}
\begin{semantics}

> The input argument $u$ is a component expression (\cref{def:component-expression}), that variable is then a clocked state variable, or a parameter expression.

Reviewer 2: I get the point, but I think it needs to be formulated more carefully to avoid ambiguity for parameter component expressions:
```suggestion
The input argument $u$ is a component expression (\cref{def:component-expression}) or a parameter expression.
In case of a component expression which is not a parameter, $u$ is a clocked state variable according to the definition above.
```

Reviewer 2: Clear in current formulation.  Resolving.
